src/components/NavigationBar.js

```diff
@@ -1,39 +1,59 @@
 import { getUserId } from 'utils';
 import logo from 'images/deploygram-logo.png';
 
 const NavigationBar = () => {
-    const navigate = useNavigate();
-    const userId = getUserId();
+  const navigate = useNavigate();
+  const userId = getUserId();
+  const [open, setOpen] = React.useState(false);
 
-    return (
-        <Box height="80px" display="flex" paddingX={3} alignItems="center" borderBottom="1px solid #dadada" bgcolor="white">
-            <Stack direction="row" justifyContent="space-between" width="100%" marginX="auto">
-                <Box sx={{ cursor: 'pointer' }} onClick={() => navigate('/')}>
-                    <img src={logo} alt="logo" width="183px"/>
-                </Box>
-                <Stack direction="row" spacing={1} alignItems="center">
-                    <IconButton aria-label="home" size="large" onClick={() => navigate('/')}>
-                        <HomeOutlinedIcon sx={{ fontSize: '43px', color: 'black' }} />
-                    </IconButton>
-                    <IconButton aria-label="create post" size="large" onClick={() => console.log('Open modal for post adding!')}>
-                        <AddBoxOutlinedIcon sx={{ fontSize: '43px', color: 'black' }} />
-                    </IconButton>
-                    <IconButton aria-label="profile" size="large" onClick={() => navigate(`/profiles/${userId}`)}>
-                        <AccountCircleIcon sx={{ fontSize: '43px', color: 'black' }} />
-                    </IconButton>
-                </Stack>
-            </Stack>
+  return (
+    <Box height="80px" display="flex" paddingX={3} alignItems="center" borderBottom="1px solid #dadada" bgcolor="white">
+      <Stack direction="row" justifyContent="space-between" width="100%" marginX="auto">
+        <Box sx={{ cursor: 'pointer' }} onClick={() => navigate('/')}>
+          <img src={logo} alt="logo" width="183px" />
+        </Box>
+        <Stack direction="row" spacing={1} alignItems="center">
+          <IconButton aria-label="home" size="large" onClick={() => navigate('/')}>
+            <HomeOutlinedIcon sx={{ fontSize: '43px', color: 'black' }} />
+          </IconButton>
+          <IconButton aria-label="create post" size="large" onClick={() => setOpen(true)}>
+            <AddBoxOutlinedIcon sx={{ fontSize: '43px', color: 'black' }} />
+          </IconButton>
+          <IconButton aria-label="profile" size="large" onClick={() => navigate(`/profiles/${userId}`)}>
+            <AccountCircleIcon sx={{ fontSize: '43px', color: 'black' }} />
+          </IconButton>
+        </Stack>
+      </Stack>
+      <Modal
+        open={open}
+        onClose={() =>setOpen(false)}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+        sx={{margin: 'auto'}}
+      >
+        <Box style={{height: "400px", width: '400px', backgroundColor: 'white', margin: '8% 35%', borderRadius: '10px', border: '1px outset black'}}>
```

Comment:
Please don't center modal content using percentage margins, maybe you could display modal as flex and center its children using justifyContent and alignItems combination?